File: app/src/main/java/com/yc/emotion/home/utils/ToastUtils.java
package com.yc.emotion.home.utils;

import android.text.TextUtils;
import android.view.Gravity;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.yc.emotion.home.R;
import com.yc.emotion.home.base.YcApplication;


public class ToastUtils {

    private static TextView sMTv_text;
    private static Toast centerToast;

    public static void showCenterToast(String text) {
        if (null==centerToast) {
            centerToast = new Toast(YcApplication.getInstance());
            centerToast.setDuration(Toast.LENGTH_LONG);
            centerToast.setGravity(Gravity.NO_GRAVITY, 0, 0);
            View view = View.inflate(YcApplication.getInstance(), R.layout.toast_center_layout, null);
            sMTv_text = view.findViewById(R.id.tv_text);
            sMTv_text.setText(TextUtils.isEmpty(text)?"null":text);
            centerToast.setView(view);
        } else {
            sMTv_text.setText(TextUtils.isEmpty(text)?"null":text);
        }
        centerToast.show();
    }
}

File: app/src/main/java/com/yc/emotion/home/utils/ToastUtils.kt
package com.yc.emotion.home.utils

import android.text.TextUtils
import android.view.Gravity
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import com.yc.emotion.home.R
import com.yc.emotion.home.base.EmApplication

object ToastUtils {
    private var sMTv_text: TextView? = null
    private var centerToast: Toast? = null
    private var ivOk: ImageView? = null

    @JvmOverloads
    fun showCenterToast(text: String?, showImg: Boolean = false) {
        if (null == centerToast) {
            centerToast = Toast(EmApplication.instance)
            centerToast?.duration = Toast.LENGTH_SHORT
            centerToast?.setGravity(Gravity.CENTER, 0, 0)
            val view = View.inflate(EmApplication.instance, R.layout.toast_center_layout, null)
            ivOk = view.findViewById(R.id.iv_ok)
            sMTv_text = view.findViewById(R.id.tv_text)
            sMTv_text?.text = if (TextUtils.isEmpty(text)) "null" else text
            ivOk?.visibility = if (showImg) View.VISIBLE else View.GONE
            centerToast?.view = view
        } else {
            sMTv_text?.text = if (TextUtils.isEmpty(text)) "null" else text
            ivOk?.visibility = if (showImg) View.VISIBLE else View.GONE
        }
        centerToast?.show()
    }
}